src/client/dom/browsing_context.hpp

```diff
@@ -23,18 +23,23 @@ namespace dom
      *
      * @param url The url of the document.
      * @param type The type of the document.
+     * @param input_type The source of the document.
      * @returns The created document.
      */
     template <typename DocumentType>
-    shared_ptr<DocumentType> create(const std::string &url, DOMParsingType type)
+    shared_ptr<DocumentType> create(const std::string &url, DOMParsingType type, InputType input_type)
     {
       shared_ptr<DocumentType> document;
       if (type == DOMParsingType::HTML)
         document = make_shared<DocumentType>(getSharedPtr<BrowsingContext>(), true);
       else
         throw std::runtime_error("Unsupported document type");
-
-      document->setUrl(url);
+      if (input_type == InputType::Text)
```

Comment:
```suggestion
      if (input_type == InputType::Source)
```
改成 Source 会好一些，和下面的方法正好能对应上